src/r1cs/mod.rs

```diff
@@ -1,4 +1,7 @@
 //! This module defines R1CS related types and a folding scheme for Relaxed R1CS
+pub mod sparse;
```

Comment:
Does this need to be public?

Reply:
The `SparseMatrix` needs to be. I've added 5e456197d9f1c204c46f0f5b0a26161107260b4c which reduces visibility to that type.

Reply:
Thanks! Why should users of Nova have public access to SparseMatrix?

Reply:
Switched to `pub(crate)`

Reply:
Thanks!